menuconfig fail on idf 4.0

In order to make the lib on IDF 4.0.1, I cloned this branch, but got errors:
```

CMake Error at C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_401/tools/cmake/component.cmake:280 (foreach):
  Syntax error in cmake code at

    C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_401/tools/cmake/component.cmake:280

  when parsing string

    C:/Users/F70/000-KL/FarmControlRoom/ESP32_TFT_library/main;C:\Users\F70\000-KL\FarmControlRoom\ESP_IDF_401/components

  Invalid character escape '\U'.
Call Stack (most recent call first):
  C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_401/tools/cmake/component.cmake:447 (__component_add_include_dirs)
  main/CMakeLists.txt:2 (idf_component_register)


-- Configuring incomplete, errors occurred!
See also "C:/Users/F70/000-KL/FarmControlRoom/ESP32_TFT_library/build/CMakeFiles/CMakeOutput.log".
cmake failed with exit code 1
```
not sure where to go from here.
Log for reference.
I've recently moved from IDF 3.3 to 4.0.1. On 3.3 I've built the master branch sucessfully with the demo app. I am not too deep into the build mechanics and git yet, accepting recommendations. system: win7
Regards, F99
[CMakeOutput.log](https://github.com/jeremyjh/ESP32_TFT_library/files/4745540/CMakeOutput.log)

Yes, I checked that first with the hello_world example. Also, I use esp-adf (on top of esp-idf) and the play_mp3 sample worked as well. If you want me to do more testing, I'll do that. 
Thanks for your quick response!
--> at the moment my best guess is, I did something non-standard with the paths, on the other hand, it's about` idf_component_register` in `main/CMakeLists.txt`
Here, you have "`$ENV{IDF_PATH}/components`" , which possibly somehow expands to 
` C:/Users/F70/000-KL/FarmControlRoom/ESP32_TFT_library/main;C:\Users\F70\000-KL\FarmControlRoom\ESP_IDF_401/components`
which contains the '`\U`' as part of '`C:\Users`'
which in turn mixed up two different directory naming styles that seemingly can't work together in the make context. Really, I don't know how to resolve the Issue, please advise.

I paste the terminal output before the error here for reference:
```
C:\Users\F70\000-KL\FarmControlRoom\ESP32_TFT_library>idf.py menuconfig
Checking Python dependencies...
Python requirements from C:\Users\F70\000-KL\FarmControlRoom\ESP_IDF_401\requirements.txt are satisfied.
Executing action: menuconfig
Running cmake in directory c:\users\f70\000-kl\farmcontrolroom\esp32_tft_library\build
Executing "cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 --warn-uninitialized -DWARN_UNINITIALIZED=1 -DCCACHE_ENABLE=1 c:\users\f70\000-kl\farmcontrol
room\esp32_tft_library"...
Warn about uninitialized values.
-- Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.21.0.windows.1")
-- mconf-idf version mconf-v4.6.0.0-idf-20190628-win32
-- IDF_TARGET not set, using default target: esp32
-- ccache will be used for faster recompilation
-- The C compiler identification is GNU 8.2.0
-- The CXX compiler identification is GNU 8.2.0
-- The ASM compiler identification is GNU
-- Found assembler: C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_TOOLS/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc.exe
-- Check for working C compiler: C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_TOOLS/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/bin/xtensa-esp32-elf
-gcc.exe
-- Check for working C compiler: C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_TOOLS/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/bin/xtensa-esp32-elf
-gcc.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_TOOLS/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/bin/xtensa-esp32-e
lf-g++.exe
-- Check for working CXX compiler: C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_TOOLS/tools/xtensa-esp32-elf/esp-2019r2-8.2.0/xtensa-esp32-elf/bin/xtensa-esp32-e
lf-g++.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Project version: a06a578
-- Building ESP-IDF components for target esp32
Loading defaults file C:/Users/F70/000-KL/FarmControlRoom/ESP32_TFT_library/sdkconfig.defaults...
C:/Users/F70/000-KL/FarmControlRoom/ESP32_TFT_library/sdkconfig.defaults:9 CONFIG_MAIN_TASK_STACK_SIZE was replaced with CONFIG_ESP_MAIN_TASK_STACK_SIZE
C:/Users/F70/000-KL/FarmControlRoom/ESP32_TFT_library/sdkconfig.defaults:10 CONFIG_TASK_WDT_TIMEOUT_S was replaced with CONFIG_ESP_TASK_WDT_TIMEOUT_S
-- Found PythonInterp: C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_TOOLS/python_env/idf4.0_py3.7_env/Scripts/python.exe (found version "3.7.3")
-- Could NOT find Perl (missing: PERL_EXECUTABLE)
-- Adding linker script C:/Users/F70/000-KL/FarmControlRoom/ESP32_TFT_library/build/esp-idf/esp32/esp32_out.ld
-- Adding linker script C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_401/components/esp32/ld/esp32.project.ld.in
-- Adding linker script C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_401/components/esp32/ld/esp32.peripherals.ld
-- Adding linker script C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_401/components/esp_rom/esp32/ld/esp32.rom.ld
-- Adding linker script C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_401/components/esp_rom/esp32/ld/esp32.rom.libgcc.ld
-- Adding linker script C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_401/components/esp_rom/esp32/ld/esp32.rom.syscalls.ld
-- Adding linker script C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_401/components/esp_rom/esp32/ld/esp32.rom.newlib-data.ld
-- Adding linker script C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_401/components/esp_rom/esp32/ld/esp32.rom.newlib-funcs.ld
CMake Error at C:/Users/F70/000-KL/FarmControlRoom/ESP_IDF_401/tools/cmake/component.cmake:280 (foreach):
```

--- a/main/CMakeLists.txt
+++ b/main/CMakeLists.txt
@@ -1,10 +1,9 @@
 set(SOURCES tft_demo.c)
 idf_component_register(
         SRCS ${SOURCES}
-        INCLUDE_DIRS
-          ${CMAKE_CURRENT_LIST_DIR}
-          $ENV{IDF_PATH}/components
         REQUIRES
             tft
             spiffs
+        PRIV_REQUIRES
+            nvs_flash
 )

--- a/main/tft_demo.c
+++ b/main/tft_demo.c
@@ -24,7 +24,7 @@
 #include "freertos/event_groups.h"
 #include "esp_sntp.h"
 #include "esp_log.h"
-#include "nvs_flash/include/nvs_flash.h"
+#include "nvs_flash.h"
 
 #endif
 

